docs/notebooks/Poincare Evaluation.ipynb

```diff
@@ -0,0 +1,1553 @@
+      "remote: Counting objects: 96, done.\u001b[K\n",
+      "remote: Total 96 (delta 0), reused 0 (delta 0), pack-reused 96\u001b[K\n",
+      "Unpacking objects: 100% (96/96), done.\n",
+      "Checking connectivity... done.\n"
+     ]
+    }
+   ],
+   "source": [
+    "% cd {parent_directory}\n",
+    "\n",
+    "# Clone repos\n",
+    "np_repo_name = 'poincare-np-embedding'\n",
+    "if force_setup and os.path.exists(np_repo_name):\n",
+    "    ! rm -rf {np_repo_name}\n",
+    "if not os.path.exists(np_repo_name):\n",
+    "    ! git clone https://github.com/nishnik/poincare_embeddings.git {np_repo_name}\n",
+    "\n",
+    "cpp_repo_name = 'poincare-cpp-embedding'\n",
+    "if force_setup and os.path.exists(cpp_repo_name):\n",
+    "    ! rm -rf {cpp_repo_name}\n",
+    "if not os.path.exists(cpp_repo_name):\n",
+    "    ! git clone https://github.com/TatsuyaShirakawa/poincare-embedding.git {cpp_repo_name}\n",
+    "\n",
+    "patches_applied = False"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 112,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-np-embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Apply patches\n",
+    "if not patches_applied:\n",
+    "    % cd {cpp_repo_name}\n",
+    "    ! git apply ../poincare_burn_in_eps.patch\n",
+    "\n",
+    "    % cd ../{np_repo_name}\n",
+    "    ! git apply ../poincare_numpy.patch\n",
+    "    \n",
+    "    patches_applied = True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 124,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding/work\n",
+      "-- Configuring done\n",
+      "-- Generating done\n",
+      "-- Build files have been written to: /home/jayant/projects/gensim/docs/notebooks/poincare/poincare-cpp-embedding/work\n",
+      "[100%] Built target poincare_embedding\n",
+      "/home/jayant/projects/gensim/docs/notebooks\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Compile the code for the external c++ implementation into a binary\n",
+    "% cd {parent_directory}/{cpp_repo_name}\n",
+    "! mkdir -p work\n",
+    "% cd work\n",
+    "! cmake ..\n",
+    "! make\n",
+    "% cd {current_directory}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You might need to install an updated version of `cmake` to be able to compile the source code. Please make sure that the binary `poincare-embedding` has been created before proceeding."
```

Comment:
Please don't ask the user to do something that's trivial to do automatically. Just check if the file exists and raise an exception if it doesn't. Otherwise, the execution flow continues and fails much later. At that point it's no longer obvious what the root cause is.

Reply:
Done